src/plugins.ts

```diff
@@ -366,4 +348,22 @@ export default class Plugins {
     )
     return plugins
   }
+
+  private isValidPlugin(p: Config): boolean {
+    if (p.valid) return true
+
+    if (!this.config.plugins.find(p => p.name === '@oclif/plugin-legacy') ||
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore check if this plugin was loaded by `plugin-legacy`
+      p._base.includes('@oclif/plugin-legacy')
```

Comment:
here we need to check if the plugin being linked/installed was loaded using the `PluginLegacy` class in the `plugin-legacy` init hook (which should be included in the CLI, not the plugin):
https://github.com/oclif/plugin-legacy/blob/main/src/hooks/init.ts#L11

This should be done using `instanceof PluginLegacy` but we would have to include `plugin-legacy` as a dependency of `plugin-plugins`, so instead just check if `p._base` includes `@oclif/plugin-legacy`.

`_base` is set in oclif/core when creating an instance of `Plugin` here:
https://github.com/oclif/core/blob/a18803454449986e03549838ba3928c66f98da37/src/config/plugin.ts#L98

but plugins loaded by `plugin-legacy` hook will have it set to `@oclif/plugin-legacy@<version>`:
https://github.com/oclif/plugin-legacy/blob/main/src/index.ts#L44